pkg/apparchive/helm-v1beta2.go

```diff
@@ -0,0 +1,288 @@
+	for _, helmChart := range helmCharts {
+		chartDir := path.Join(helmDir, helmChart.GetDirName())
+		if err := os.MkdirAll(chartDir, 0744); err != nil {
+			return errors.Wrap(err, "failed to create chart dir")
+		}
+
+		archive, err := base.FindHelmChartArchiveInRelease(u.Files, helmChart)
+		if err != nil {
+			return errors.Wrap(err, "failed to find helm chart archive in release")
+		}
+
+		archivePath := path.Join(chartDir, fmt.Sprintf("%s-%s.tgz", helmChart.Spec.Chart.Name, helmChart.Spec.Chart.ChartVersion))
+		if err := ioutil.WriteFile(archivePath, archive, 0644); err != nil {
+			return errors.Wrap(err, "failed to write helm chart archive")
+		}
+
+		mergedValues := helmChart.Spec.Values
+		if mergedValues == nil {
+			mergedValues = map[string]kotsv1beta2.MappedChartValue{}
+		}
+		for _, optionalValues := range helmChart.Spec.OptionalValues {
+			parsedBool, err := strconv.ParseBool(optionalValues.When)
+			if err != nil {
+				return errors.Wrap(err, "failed to parse when conditional on optional values")
+			}
+			if !parsedBool {
+				continue
+			}
+			if optionalValues.RecursiveMerge {
+				mergedValues = kotsv1beta2.MergeHelmChartValues(mergedValues, optionalValues.Values)
+			} else {
+				for k, v := range optionalValues.Values {
+					mergedValues[k] = v
+				}
+			}
+		}
+
+		helmValues, err := helmChart.Spec.GetHelmValues(mergedValues)
+		if err != nil {
+			return errors.Wrap(err, "failed to get local values for chart")
+		}
+
+		valuesContent, err := yaml.Marshal(helmValues)
+		if err != nil {
+			return errors.Wrap(err, "failed to marshal values")
+		}
+
+		builder, _, err := base.NewConfigContextTemplateBuilder(u, renderOptions)
+		if err != nil {
+			return errors.Wrap(err, "failed to create config context template builder")
+		}
+
+		renderedValuesContent, err := builder.RenderTemplate(fmt.Sprintf("%s-values", helmChart.GetDirName()), string(valuesContent))
+		if err != nil {
+			return errors.Wrap(err, "failed to render values")
+		}
+
+		valuesPath := path.Join(chartDir, "values.yaml")
+		if err := ioutil.WriteFile(valuesPath, []byte(renderedValuesContent), 0644); err != nil {
+			return errors.Wrap(err, "failed to write values file")
+		}
+
+		builderValues := helmChart.Spec.Builder
+		if builderValues == nil {
+			builderValues = map[string]kotsv1beta2.MappedChartValue{}
+		}
+
+		builderHelmValues, err := helmChart.Spec.GetHelmValues(builderValues)
+		if err != nil {
+			return errors.Wrap(err, "failed to get builder values for chart")
+		}
+
+		builderValuesContent, err := yaml.Marshal(builderHelmValues)
+		if err != nil {
+			return errors.Wrap(err, "failed to marshal builder values")
+		}
+
+		renderedBuilderValuesContent, err := builder.RenderTemplate(fmt.Sprintf("%s-builder-values", helmChart.GetDirName()), string(builderValuesContent))
+		if err != nil {
+			return errors.Wrap(err, "failed to render builder values")
+		}
+
+		builderValuesPath := path.Join(chartDir, "builder-values.yaml")
```

Comment:
instead of storing it in the `helm` directory where it's not really relevant for the user to see, can we move rendering the builder values to be in the same place where processing the images happen?

Reply:
yes. as discussed, going to move the image processing to the midstream flow.  this is when the helm archive will be written to the `helm` directory and then it will be rendered with builder values to process images